src/control/server/mgmt_pool.go

```diff
@@ -171,15 +173,20 @@ func (svc *mgmtSvc) calculateCreateStorage(req *mgmtpb.PoolCreateReq) error {
 	if len(req.GetRanks()) == 0 {
 		return errors.New("zero ranks in calculateCreateStorage()")
 	}
-	if req.MetaBlobBytes > 0 && !instances[0].GetStorage().BdevRoleMetaConfigured() {
-		return errors.New("meta size set in request but md-on-ssd is not enabled in config")
+
+	mdOnSSD := instances[0].GetStorage().BdevRoleMetaConfigured()
+	switch {
+	case !mdOnSSD && req.MemRatio > 0:
+		// Prevent MD-on-SSD parameters being used in incompatible mode.
+		return FaultPoolMemRatioNoRoles
+	case mdOnSSD && req.MemRatio == 0:
+		// Set reasonable default if not set in MD-on-SSD mode.
+		req.MemRatio = storage.DefaultMemoryFileRatio
```

Comment:
A notification message should probably be printed indicating that the default memory ratio is going to be used.

Reply:
updated in https://github.com/daos-stack/daos/pull/14844